app/controllers/concerns/client_sortable.rb

```diff
@@ -24,38 +24,52 @@ def filtered_and_sorted_clients(default_order: nil)
 
   # see if there are any overlapping keys in the provided params and search/sort set
   def has_search_and_sort_params?
-    (params.keys.map(&:to_sym) & search_and_sort_params).any?
+    overlapping_keys = (params.keys.map(&:to_sym) & search_and_sort_params)
+    hash_params = params.try(:to_unsafe_h) || params
+    overlapping_keys.any? && hash_params.slice(*overlapping_keys).any? { |_, v| v.present? }
   end
 
   private
 
   def setup_sortable_client
-    reset_filter_params if params[:clear]
+    delete_cookie if params[:clear]
```

Comment:
I'd love to see a unit spec for these cookie-related changes , perhaps in a `"with filter cookies"` context, calling  `fitered_and_sorted_clients`.